src/lib/crawler/crawler.ts

```diff
@@ -1,19 +1,47 @@
 import axios from "axios";
+import * as cheerio from "cheerio";
 
+// defining Crawler interface
 export interface Crawler {
   getMatchIds(params: { character: string; realm: string }): Promise<string[]>;
-  getMatchDetails(params: { matchIds: string[] }): Promise<any[]>;
+  getMatchDetails(params: { matchIds: string[] }): Promise<any[]>; //
 }
-
+// implementing the Crawler interface in the WarmaneCrawler class
 export class WarmaneCrawler implements Crawler {
+  // defining 'getMatchIds' method with optional paramater for testing (html)
   async getMatchIds(params: {
     character: string;
     realm: string;
+    // added optional paramater (HTML passed as string) for testing
+    html?: string;
   }): Promise<string[]> {
-    // throw new Error("Method not implemented.");
-    return ["1", "2", "3"];
+    // getMatchId will use optional parameter if present, will only use URL in its absence
+    const html =
+      params.html ??
+      (await axios
+        .get(
+          `https://armory.warmane.com/character/${params.character}/${params.realm}/match-history`
+        )
+        .then((response) => response.data));
```

Comment:
So this works, but I personally prefer to not combine `await` `.then` `.catch` or `.finally`, in one expression. 

To be honest, with the advent of `async/await` I actually try to never use `.then` `.catch` or `.finally` in production code.

I recommend:
```TypeScript
  const response = await axios.get<string>(
    `https://armory.warmane.com/character/${params.character}/${params.realm}/match-history`
  );
  const html: string = response.data
```
This is TypeScript thing that might be new to you but the `string` part in  `axios.get<string>()`, is a type parameter. It's basically saying, "The data property on the response object will be of type `string`"